### ISSUE TITLE
20171009-20171015：TensorFlow的学习

### ISSUE THREAD

**Issue by stuian** [Neuqmiao/deepNLP#1]
## 此issue为本周提交作业以及提问讨论的区域。

提交作业的形式为：
- 姓名：xxx
- 作业：链接
- 感想：可以对本周学习内容进行小总结，或者学习过程中的感受，也可以是学习过程中的问题等等。

**Comment by WALLE499** [comment]
赵松：
作业：https://github.com/WALLE499/Hello-AI.git
总结：1、tensorflow的初级认识和使用。对tf的函数不了解，目前只学习了几个函数。
           2、通过MNIST的例子大概了解了Tensorflow的运行方式。目前还没有尝试过复杂的结构。
           3、TensorFlow与我们正常的编程思维略有不同：TensorFlow中的语句不会立即执行；而是等到开启会话session的时候，才会执行session.run()中的语句。如果run中涉及到其他的节点，也会执行到。
          4、Tesorflow模型中的所有的节点都是可以视为运算操作op或tensor。
          5、需要多练习，接下来练习MNIST进阶以及更复杂的神经网络。
问题：1、Session会话控制不太了解，是不是执行哪一步就需要进行sess.run()?

**Comment by Yukiococoa** [comment]
姓名：田雪骐
作业：https://github.com/Yukiococoa/Tensorflow
感想：
1.前期主要在熟悉Python语法以及TensorFlow库的函数
2.运行机制和RDD比较像，调用run函数时才执行前面定义的语句
3.占位符Placeholder挺好用的，很巧妙
问题：
1.可视化框架Tensorboard一直打不开..
2.要接触的新函数不少，CNN的例子遇到很多不明白的问题，初步认为是基本语法问题，考虑先过一遍Python

**Comment by MH-lxj** [comment]
李雪晶：
作业：https://github.com/MH-lxj/deepNLP/tree/patch-3
总结：
1、我在写代码的时候输入print也会报错，所以python3中要在print后输出的内容加括号
2、写代码是tf.sub和tf.mul都会报错，这里不能简写，应写成tf.subtract()和tf.multiply()
3、 tf.placeholder(tf.types.float32) 会报错，所以type应该去掉
4、相对应的MNIST数据集的标签是介于0到9的数字，用来描述给定图片里表示的数字。我们使标
签数据是"one-hot vectors"。 一个one-hot向量除了某一位的数字是1以外其余各维度数字都是0。所以
数字n将表示成一个只有在第n维度（从0开始）数字为1的10维向量。这个很巧妙的处理方法。
5、做了tensorflow环境的配置，看来几个关于tensorflow的视频，中文教程看了第一章。看了MNIST，但是还没有仔细看代码，接下来打算继续照着中文版学习。
问题：
1、MNIST中：
  为了得到一张给定图片属于某个特定数字类的证据（evidence），我们对图片像素值进行加权求和。如果这个像素具有很强的证据说明这张图片不属于该类，那么相应的权值为负数，相反如果这个像素拥有有利的证据支持这张图片属于这个类，那么权值是正数。这个权值怎么确定？
2、softmax回归模型图中，权重只体现输出数字一二三、但是还可以四五六七八九啊？不是0-9的数字都有可能？

**Comment by ylwxy2015** [comment]
姓名：王鑫禹
作业：https://github.com/ylwxy2015/tensor_test
感想：
1.有tensorflow这类工具的存在大大降低了DL的学习门槛，官方的教程和莫烦的教程简单易懂
2.tensorflow里有一些与numpy功能类似的库函数，目前还不能体会之间的细微区别。还不能得心应手地使用tensor flow里的库函数
3.通过此次tensorflow的学习得知了keros，kaffe等的存在，目前不仅是理论上，在工具掌握上的广度与深度都有待提升，这也是今后学习的目标。
问题：
1.莫烦教程中的tensorboard功能不能正常运行，一起学习的小伙伴却能成功运行……
2.还有一些CNN中的细节问题比如“官方中文教程2.4节中W_conv2=weight_variable([5,5,32,64]) 32意味着？”CNN的理论理解得还不够透彻，代码的学习也是对理论的复习

**Comment by Lucas-Fan** [comment]
姓名：樊志英
作业：https://github.com/Lucas-Fan/Tensorflow   （一个小时代码还没上传成功，先把网址打上明天再试试）
感想：
1.第一次详细学习一个框架，感觉Tensorflow还是很方便的有很多的功能，虽然新的版本感觉有些臃肿。
2.它和python一样总是更新，要注意版本问题。
3.感觉用Tensorflow实现cnn和rnn后，对于rnn和cnn有了不一样的理解。
问题：
1.不知道tensorboard那个图怎么看，还没来得看文档。
2.还是要自己多次敲代码建立新的神经网络模型，才能对Tensorflow更深入的理解。

**Comment by stuian** [comment]
姓名：蒋文君
作业笔记：[https://github.com/stuian/NLP/blob/master/lec8/note/tensorflow.md](https://github.com/stuian/NLP/blob/master/lec8/note/tensorflow.md)
感想：

- 原来学理论一直没跟上，这次写代码了，终于明白了一点点，原来理解的输入和输出都错了。
- 目前只是简单了解了tensorflow的语法，知道三层神经网络怎么编写，mnist还没运行，不知道数据怎么导入和处理。

问题：
Mnist数据集怎么处理的没明白，手动解压？

**Comment by Sicanno** [comment]
姓名：刘云珂
作业笔记：https://github.com/Sicanno/tensorflow/blob/master/first.md
感想：
* 这次主要安装软件花了一部分时间，所以对于建立网络还未熟悉。
* 简单了解了一些python tensorflow的语法，可以建立一个简单的神经网络。
* 莫烦的视频简单易懂。<br>
问题：
tensorboard最后一步网址打不开

**Comment by Lonny97** [comment]
姓名：王历宁
作业：//还不太会用git。。
学习内容
本周主要学习了tensorflow的基本使用方法及cnn原理和数据可视化，具体包括1.pandas库的使用2.matplotlib的使用3.seaborns的使用
问题：
1.池化 及Dropout参数设置还没搞清楚
2.tensorflow学习时间感觉还不够，使用不熟练

**Comment by guangyaooo** [comment]
姓名：丁光耀
作业：https://github.com/Dgy2017/tf_1.git
感想:
本周主要了解了一下tensorflow使用方法，看了官方给的几个手写数字识别的例子，还有很多tensorflow的函数需要去学习，主要是不明白其调用结果及什么时候调用
问题：
- cnn网络的卷积层具体做了什么事情

**Comment by zjp96machineL** [comment]
姓名：张金鹏
作业：https://github.com/zjp96machineL/-.git
感想：
这周主要学习的内容是tensorflow的基本流程，以及最基本的用法。其中，基本用法主要是包括，Session，Variable，placeholder，激励函数，dropout等。而基本流程则大体为
1.训练数据
2.定义节点准备接收数据
3.定义神经层：隐藏层和预测层
4.定义 loss 表达式
5.选择 optimizer并训练 使 loss 达到最小
这样一个形式。
存在问题：如何将一个文本数据集读入并且进行预处理还没有搞明白，只是实现了简单的RNN结构，还有部分参数设置不明白。

**Comment by key000** [comment]
姓名：李园溪
作业：这个还不太会用
感想
学习了安装tensorflow以及相关操作
学习了简单的python的一些语法和简单例子
问题：
还没有熟悉网络相关
tensorflow只是基本会用，并不熟悉
github没完全会用

**Comment by hzc5095** [comment]
姓名：郝增超
作业：不会用啊，怎么弄自己的文档。
感想：1.安装软件过程有点艰辛，说明对一些计算机的错误还不太熟悉。
2.简单学习了tensorflow的一些基本知识，但是更深的用法还没有看到。
3.使用tensorflow时候因为Python版本是3.5，和2.5版本语法上有一定的区别，使用时应注意。
